### PR TITLE
Allow running multiple Upstreams in parallel

### PR DESCRIPTION
**Pull request by rudolfs**
For testing purposes.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -114,6 +114,29 @@ We use [Cypress](https://www.cypress.io/) for integration tests and
...
+At the moment this is only possible in development mode.
+
+```
+mkdir /Users/rudolfs/work/19000
+mkdir /Users/rudolfs/work/20000
+
+# Launch the first instance.
+
+RAD_HOME="/Users/rudolfs/work/19000" \
+RADICLE_UPSTREAM_PROXY_ARGS="--http-listen 127.0.0.1:19000 --peer-listen 0.0.0.0:19000" \
```
Comment (FintanH) [inline DEVELOPMENT.md:128]: I'd also just let the port be chosen as well
```suggestion
RADICLE_UPSTREAM_PROXY_ARGS="--http-listen 127.0.0.1:19000 --peer-listen 0.0.0.0:0" \
```

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -114,6 +114,29 @@ We use [Cypress](https://www.cypress.io/) for integration tests and
...
+
+At the moment this is only possible in development mode.
+
+```
+mkdir /Users/rudolfs/work/19000
+mkdir /Users/rudolfs/work/20000
+
+# Launch the first instance.
+
+RAD_HOME="/Users/rudolfs/work/19000" \
```
Comment (FintanH) [inline DEVELOPMENT.md:127]: Note you could also just use `RAD_PROFILE` which is what I did. Created two profiles and then started them with env var. Could be worth documenting here too

File: DEVELOPMENT.md (hunk 1 above)
Comment (rudolfs) [reply DEVELOPMENT.md:128]: If you do that it'll be harder to link them up if you want to test replication locally. You'll need to look at the logs to figure out on which port it bound. But I'll mention this possibility in the docs.

Comment (rudolfs) [reply DEVELOPMENT.md:129]: bd7da77777fd597559eb25c23b8521b0bafba86b

File: DEVELOPMENT.md (hunk 2 above)
Comment (rudolfs) [reply DEVELOPMENT.md:128]: f14408ec8d3bdf0f9d0dcb74c444a989c7b52414

**Review by xla — APPROVED** [review]
⤵️ 👱 🏌 🎿

**Review by MeBrei — APPROVED** [review]

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -114,6 +114,35 @@ We use [Cypress](https://www.cypress.io/) for integration tests and
...
+# And then in a separate shell, launch the second instance.
+
+RAD_HOME="/Users/rudolfs/work/20000" \
+RADICLE_UPSTREAM_PROXY_ARGS="--http-listen 127.0.0.1:20000 --peer-listen 0.0.0.0:20000" \
+RADICLE_UPSTREAM_UI_ARGS="backend=localhost:20000" \
+yarn start
+```
+
+It is possible to let the OS choose the peer port by setting it to:
+`--peer-listen 0.0.0.0:0`. And if you don't need completely isolated state,
```
Comment (FintanH) [inline DEVELOPMENT.md:142]: They're still completely isolated, no? :)

**Review by FintanH — APPROVED** [review]

File: DEVELOPMENT.md (hunk 3 above)
Comment (rudolfs) [reply DEVELOPMENT.md:142]: No, electron state is shared when using `RAD_PROFILE`.

**Comment by rudolfs** [comment]
Merged via: d13a8089dfbf087c4fff575c81569e50b7f172a0.

**Pull request closed without merging**